From 3e03ab933f4824d947be39d4c2ed8d8a65b77020 Mon Sep 17 00:00:00 2001
From: Patrick Doyle <pdoyle@irobot.com>
Date: Fri, 21 Jun 2019 14:41:06 -0400
Subject: [PATCH] Tweak irobot-mt7688.h to meet our needs:

- Add support for booting images up to 32MB uncompressed.
- Use UART1 for the SPL instead of UART0
- Increase malloc size from 1M to 16M
---
 include/configs/irobot-mt7688.h | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/include/configs/irobot-mt7688.h b/include/configs/irobot-mt7688.h
index f04a976c..9ddc133e 100644
--- a/include/configs/irobot-mt7688.h
+++ b/include/configs/irobot-mt7688.h
@@ -7,13 +7,14 @@
 
 #ifndef __CONFIG_IROBOT_MT7688_H
 #define __CONFIG_IROBOT_MT7688_H
+#include <linux/sizes.h>
 
 #define CONFIG_SYS_HZ			1000
 #define CONFIG_SYS_MIPS_TIMER_FREQ	290000000
 
 #define CONFIG_SYS_MONITOR_BASE		CONFIG_SYS_TEXT_BASE
 
-#define CONFIG_SYS_MALLOC_LEN		0x100000
+#define CONFIG_SYS_MALLOC_LEN		0x1000000
 #define CONFIG_SYS_BOOTPARAMS_LEN	0x20000
 
 #define CONFIG_SYS_SDRAM_BASE		0x80000000
@@ -21,7 +22,8 @@
 
 #define CONFIG_SYS_INIT_SP_OFFSET	0x80000
 
-#define CONFIG_SYS_BOOTM_LEN		0x1000000
+/* Allow uncompressed images up to 32MB (of our 64) */
+#define CONFIG_SYS_BOOTM_LEN		SZ_32M
 
 #define CONFIG_SYS_MAXARGS		16
 #define CONFIG_SYS_CBSIZE		1024
@@ -41,8 +43,8 @@
 #define CONFIG_SYS_NS16550_MEM32
 #define CONFIG_SYS_NS16550_CLK		40000000
 #define CONFIG_SYS_NS16550_REG_SIZE	-4
-#define CONFIG_SYS_NS16550_COM1		0xb0000c00
-#define CONFIG_CONS_INDEX		1
+#define CONFIG_SYS_NS16550_COM2		0xb0000d00
+#define CONFIG_CONS_INDEX		2
 #endif
 
 /* Serial common */
-- 
2.24.1

